src/actions/importing/aioseo-posts-importing-action.php

```diff
@@ -237,14 +283,23 @@ public function index() {
 	 * @return Indexable The created indexables.
 	 */
 	public function map( $indexable, $aioseo_indexable ) {
-		// Do not overwrite any existing values.
-		foreach ( $this->aioseo_to_yoast_map as $aioseo_key => $yoast_key ) {
-			if ( ! empty( $indexable->{$yoast_key} ) ) {
+		foreach ( $this->aioseo_to_yoast_map as $aioseo_key => $yoast_mapping ) {
+			// For robots import.
+			if ( isset( $yoast_mapping['robots_import'] ) && $yoast_mapping['robots_import'] ) {
+				$yoast_mapping['subtype']                  = $indexable->object_sub_type;
+				$indexable->{$yoast_mapping['yoast_name']} = \call_user_func( [ $this, $yoast_mapping['transform_method'] ], $aioseo_indexable, $yoast_mapping );
```

Comment:
The AIOSEO metakey is duplicated in the aioseo_to_yoast_map in the array keys and as the `aioseo_key` value. Have you considered passing the `$aioseo_key` variable that is available here to the `transform_method` so you don't need to store it twice in the map?

Reply:
I did considered it and I actually wondered what's worse, having an extra map key/value pair, or having an extra argument (hence a bit more logic) to worry about.

It looks like you prefer the latter, which is a great reason for me to revisit that 😅 thanks!